code/game/objects/items.dm

```diff
@@ -428,6 +428,12 @@ var/global/objects_thrown_when_explode = FALSE
 			return
 		//user.next_move = max(user.next_move+2,world.time + 2)
 	add_fingerprint(user)
+	
+	if(src.on_fire)
```

Comment:
no src, otherwise looks fine

Reply:
i just copied the thing because there's another src.on_fire 5 lines down